src/main/java/gov/cabinetoffice/gap/adminbackend/controllers/ApplicationFormController.java

```diff
@@ -118,6 +131,46 @@ public ResponseEntity deleteApplicationForm(@PathVariable @NotNull Integer appli
 
     }
 
+    @PostMapping("/lambda/{grantAdvertId}")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "204", description = "Application form updated successfully.",
+                    content = @Content(mediaType = "application/json")),
+            @ApiResponse(responseCode = "400", description = "Bad request body",
+                    content = @Content(mediaType = "application/json")),
+            @ApiResponse(responseCode = "403",
+                    description = "Insufficient permissions to update this application form.",
+                    content = @Content(mediaType = "application/json")),
+            @ApiResponse(responseCode = "404", description = "Application not found with given id",
+                    content = @Content(mediaType = "application/json")), })
+    public ResponseEntity removeApplicationAttachedToGrantAdvert(
+            @RequestHeader(HttpHeaders.AUTHORIZATION) String authHeader, @PathVariable @NotNull UUID grantAdvertId) {
+        try {
+            secretAuthService.authenticateSecret(authHeader);
+            Integer schemeId = grantAdvertService.getAdvertById(grantAdvertId, true).getScheme().getId();
+            ApplicationFormEntity applicationForm = applicationFormService.getApplicationFromSchemeId(schemeId);
+
+            ApplicationFormPatchDTO applicationFormPatchDTO = new ApplicationFormPatchDTO();
+            applicationFormPatchDTO.setApplicationStatus(ApplicationStatusEnum.REMOVED);
+            this.applicationFormService.patchApplicationForm(applicationForm.getGrantApplicationId(),
+                    applicationFormPatchDTO, true);
+
+            return ResponseEntity.noContent().build();
+        }
+
+        catch (NotFoundException nfe) {
```

Comment:
There is no test coverage for the errors in the catch block

Reply:
Have added that now